### ISSUE TITLE
breaks on styles that have text-field set to a function

### ISSUE THREAD

**Issue by tmcw** [mapbox/mapbox-gl-style-lint#55]
testcase: https://gist.github.com/tmcw/5c435005d112528f8a9e

```
/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:192
     if (spec[l].required && spec[l]['default'] === undefined && val[l] === un
                                                                    ^
TypeError: Cannot read property 'stops' of null
    at Function.validate.object (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:192:76)
    at Function.validate.function (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:285:18)
    at validate (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:43:37)
    at validateStop (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:298:21)
    at Function.validate.array (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:220:36)
    at validate.object.stops (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:288:26)
    at Function.validate.object (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:180:50)
    at Function.validate.function (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:285:18)
    at validate (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:43:37)
    at Function.validate.object (/Users/tmcw/src/mapbox-gl-style-lint/lib/validate/parsed.js:180:50)
```

**Comment by jfirebaugh** [comment]
This is definitely breaking in a way that it shouldn't. Do we need to support null values for properties though? I.e., should `"text-field": null` work? Maybe this is necessary for `paint.foo` properties to override `paint` properties?

**Comment by jfirebaugh** [comment]
With proposed fix above, output is: "layers[0].layout.text-field.stops[1][1]: string expected, null found".

**Comment by tmcw** [comment]
Fixed in e74c38a4c61da206dccfec97d6e3b766f0bf1551
